Uplift/Views/GymDetail/GymDetailFacilitiesCell.swift

```diff
@@ -7,103 +7,89 @@
+        collectionViewLayout.minimumLineSpacing = 0
 
-    private var gymFacilities: [String] = []
+        collectionView = UICollectionView(frame: .zero, collectionViewLayout: collectionViewLayout)
 
-    override init(frame: CGRect) {
         super.init(frame: frame)
-
-        setupViews()
-        setupConstraints()
+        
+        backgroundColor = .white
+        isUserInteractionEnabled = true
+
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        collectionView.register(FacilitiesDropdownCell.self, forCellWithReuseIdentifier: Identifiers.facilitiesDropdownCell)
+        collectionView.isScrollEnabled = false
+        collectionView.backgroundColor = .white
+        contentView.addSubview(collectionView)
     }
 
-    // MARK: - Public configure
-    func configure(for gymDetail: GymDetail) {
-        gymFacilities = gymDetail.facilitiesList
-
-        DispatchQueue.main.async {
-            self.gymFacilitiesTableView.reloadData()
-            self.setupConstraints()
-        }
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
     }
 
-    func setupViews() {
-        facilitiesLabel.font = ._16MontserratBold
-        facilitiesLabel.textAlignment = .center
-        facilitiesLabel.textColor = .primaryBlack
-        facilitiesLabel.text = ClientStrings.GymDetail.facilitiesSection
-        contentView.addSubview(facilitiesLabel)
-
-        gymFacilitiesTableView = UITableView(frame: .zero, style: .plain)
-        gymFacilitiesTableView.bounces = false
-        gymFacilitiesTableView.showsVerticalScrollIndicator = false
-        gymFacilitiesTableView.separatorStyle = .none
-        gymFacilitiesTableView.backgroundColor = .clear
-        gymFacilitiesTableView.isScrollEnabled = false
-        gymFacilitiesTableView.allowsSelection = false
-        gymFacilitiesTableView.register(GymFacilitiesCell.self, forCellReuseIdentifier: GymFacilitiesCell.identifier)
-        gymFacilitiesTableView.delegate = self
-        gymFacilitiesTableView.dataSource = self
-        contentView.addSubview(gymFacilitiesTableView)
-
-        dividerView.backgroundColor = .gray01
-        contentView.addSubview(dividerView)
+    func configure(for facilities: [Facility], dropdownStatuses: [DropdownStatus], reloadGymDetailCollectionViewClosure: @escaping ((Int) -> ())) {
+        setNeedsUpdateConstraints()
+        self.facilities = facilities
+        self.dropdownCellStatuses = dropdownStatuses
+        self.reloadGymDetailCollectionViewClosure = reloadGymDetailCollectionViewClosure
+        collectionView.reloadData()
     }
 
-    func setupConstraints() {
-        facilitiesLabel.snp.makeConstraints { make in
-            make.top.equalToSuperview().inset(Constraints.verticalPadding)
-            make.centerX.equalToSuperview()
-            make.height.equalTo(Constraints.titleLabelHeight)
+    override func updateConstraints() {
```

Comment:
we shouldn't have to override `updateConstraints`, like kevin mentioned in `FacilitiesDropdownCell`